### PR TITLE
Integrate `scan-files` with duplicate S3 bucket

### PR DESCRIPTION
**Pull request by smcmurtry**
# Summary | Résumé

An alternative approach to #97. This PR integrates malware scanning via `scan-files` in the document download api. It does this by uploading each file it receives to a 2nd S3 bucket watched by `scan-files`. `scan-files` then writes the scan verdict as a tag on the S3 object. 

I have added a new endpoint that will be used by notification-api: `/services/<service_id>/documents/<document_id>/scan-verdict`. When that endpoint gets a POST request, the scan verdict for the document is checked and a 4xx response is returned if the scan is a) in progress, b) malicious, or c) suspicious. All other verdicts result in a 200 response.

For more details about how this will work with notification-api, see the [draft ADR](https://github.com/cds-snc/notification-adr/blob/feat/malicious-content-scanning/records/2023-01-24.malicious-content-scanning.md#duplicate-s3-bucket).

# Test instructions | Instructions pour tester la modification

1. Check out this branch and add `SCAN_FILES_DOCUMENTS_BUCKET=notification-canada-ca-staging-document-download-scan-files` to your .env
2. run dd-api with `make run`
3. Check out the following branch for notification-api: `feat/scan-uploaded-files-3`
4. edit your notification-api/.env file to include: `DOCUMENT_DOWNLOAD_API_HOST=http://host.docker.internal:7000`
5. run locally with `make run` and start celery.
6. Log on to AWS Staging Admin site and find the S3 bucket with "scan-files" in its name.
7. Delete all contents of this bucket. 
8. Send a file attachment via your local API.
9. Look in the "scan-files" bucket and verify that a new file has been created and it has some tags, including "av-status"
10. Verify that your email and attachment was delivered successfully, and you see an info log from celery like this:
```
scan_verdict for document_id dc3dd963-1fb6-4911-8ab6-d1c4727029e7 is {'scan_verdict': 'clean'}
```

### CODE REVIEW

**Review by andrewleith — COMMENTED** [review]
This is looking very good.  My only concern/comments are around what would happen if somehow a notification got sent where the scan verdict was a negative one.  

Given the async nature of this setup (between API, Scan files and dd-api), I wonder if we should have some code in place to prevent a user from downloading those files?

File: app/download/views.py
Comment (smcmurtry) [reply app/download/views.py:106]: Definitely a fair point - I have it set up so that notification-api will make a request to this endpoint before downloading the file. But it is probably more foolproof to do it the way you are suggesting. I will give that a try.

Comment (smcmurtry) [reply app/download/views.py:106]: Actually on second thought - I did have it set up like you are suggesting before, and the reason I moved to a single endpoint was because we have 2 endpoints for downloading files - one for "attach" mode and one for "link" mode. For link mode, the download endpoint is not actually called by notification-api, only by the user who downloads the file. So we need some kind of dedicated "check scan status" endpoint to prevent link mode from sending viruses. 

That being said, I could keep this new endpoint but also call `check_scan_verdict` from the other 2 "download document" endpoints as another layer of safety.

Comment (andrewleith) [reply app/download/views.py:106]: Makes sense to me!

Comment (smcmurtry) [reply app/download/views.py:106]: Okay as @andrewleith and I discussed offline - I will make a follow up PR to add the check to the 2 doc download endpoints, and that can be merged after changes to the notification-api are merged. That way we can merge this PR and still not cause compatibility problems with the current notification-api.

**Review by andrewleith — APPROVED** [review]
Looks great, nicely done! 🚀

File: app/upload/views.py
Comment (smcmurtry) [reply app/upload/views.py:47]: I'll address this in the follow up PR

File: app/download/views.py
Comment (jimleroyer) [inline app/download/views.py:118]: We could have these error code hardcoded in constants at the top of the module, to easily keep track and not duplicate.

File: app/download/views.py
Comment (jimleroyer) [reply app/download/views.py:118]: As a 403 is quite standard for unauthorized access, it might be a bit misleading. The API has access but we refuse to send it back. The `423 - locked` status could be more aligned, or we can use a number that isn't in use with our our definition.

**Comment by jimleroyer** [comment]
Well done!

**Pull request merged**
